src/AWS.Deploy.Orchestration/Utilities/Helpers.cs

```diff
@@ -0,0 +1,90 @@
+            {
+                throw new TimeoutException();
+            }
+        }
+
+        /// <summary>
+        /// This method returns the deployment tool workspace directory to create the CDK app during deployment.
+        /// It will first look for AWS_DOTNET_DEPLOYTOOL_WORKSPACE environment variable set by the user. It will be used as the deploy tool workspace if it points to a valid directory whithout whitespace characters in its path.
+        /// If the environment variable is set, it will also create a temp directory inside the workspace and set process scoped TEMP and TMP environment variables that point to the temp directory.
+        /// This additional configuration is required due to a known issue in the CDK - https://github.com/aws/aws-cdk/issues/2532
+        /// If the override is not present, then it defaults to USERPROFILE/.aws-dotnet-deploy.
+        /// It will throw an exception if the USERPROFILE contains a whitespace character.
+        /// </summary>
+        public static string GetDeployToolWorkspaceDirectoryRoot(string userProfilePath, IDirectoryManager directoryManager, IEnvironmentVariableManager environmentVariableManager)
+        {
+            var overridenWorkspace = environmentVariableManager.GetEnvironmentVariable("AWS_DOTNET_DEPLOYTOOL_WORKSPACE", EnvironmentVariableTarget.User)
+               ?? environmentVariableManager.GetEnvironmentVariable("AWS_DOTNET_DEPLOYTOOL_WORKSPACE", EnvironmentVariableTarget.Machine);
+
+            var defaultWorkSpace = Path.Combine(userProfilePath, ".aws-dotnet-deploy");
+
+            if (overridenWorkspace == null)
+            {
+                return !defaultWorkSpace.Contains(" ")
+                    ? defaultWorkSpace
+                    : throw new InvalidDeployToolWorkspaceException(DeployToolErrorCode.InvalidDeployToolWorkspace, $"The USERPROFILE path ({userProfilePath}) contains a whitespace character and cannot be used as a workspace by the deployment tool. " +
+                    "Please refer to the troubleshooting guide for additional configuration required to handle this issue.");
+            }
+
+            if (!directoryManager.Exists(overridenWorkspace))
+            {
+                throw new InvalidDeployToolWorkspaceException(DeployToolErrorCode.InvalidDeployToolWorkspace,
+                    $"The AWS_DOTNET_DEPLOYTOOL_WORKSPACE environment variable has been set to \"{overridenWorkspace}\" but it does not point to a valid directory.");
+            }
+
+            if (overridenWorkspace.Contains(" "))
+            {
+                throw new InvalidDeployToolWorkspaceException(DeployToolErrorCode.InvalidDeployToolWorkspace,
+                    $"The AWS_DOTNET_DEPLOYTOOL_WORKSPACE environment variable ({overridenWorkspace}) contains a whitespace character and cannot be used as a workspace by the deployment tool.");
+            }
+
+            var tempDir = Path.Combine(overridenWorkspace, "temp");
```

Comment:
Add comment about why we are setting the temp environment variable to work around CDK issue.

Reply:
Done